### PR TITLE
IsoBuffer.from(): better type checking, Typed arrays (0.33)

### PR DESCRIPTION
**Pull request by vladsud**

### CODE REVIEW

File: common/lib/common-utils/src/bufferBrowser.ts [hunk 1]
```diff
@@ -78,6 +88,12 @@ export class IsoBuffer extends Uint8Array {
     static from(value, encodingOrOffset?, length?): IsoBuffer {
         if (typeof value === "string") {
             return IsoBuffer.fromString(value, encodingOrOffset as string | undefined);
+        // Capture any typed arrays, including Uint8Array (and thus - IsoBuffer!)
+        } else if (value !== null && typeof value === "object" && isArrayBuffer(value.buffer)) {
+            // Support currently for full array, no view ports! (though it can be added in future)
+            assert(value.byteOffset === 0);
```
Comment (anthony-murphy) [inline common/lib/common-utils/src/bufferBrowser.ts:94]: we assert byteOffset  ==0 but above in isArrayBuffer we require it to be undefined

Comment (vladsud) [reply common/lib/common-utils/src/bufferBrowser.ts:94]: value.buffer.byteOffset  == undefined.
value.byteOffset === 0

Comment (vladsud) [reply common/lib/common-utils/src/bufferBrowser.ts:94]: Basically we say outer is typed array, inner is not

Comment (anthony-murphy) [reply common/lib/common-utils/src/bufferBrowser.ts:94]: oh. i see. we are looking a value.buffer above, not value itself

**Review by anthony-murphy — APPROVED** [review]

**Comment by vladsud** [comment]
Note: I missed that UT is missing here - it was added later (see "bufferToString working with IsoBuffer" case in 0.35+).
I've validated that tests are passing with these changes

**Pull request merged**